### ISSUE TITLE
Use sender signature in PM ticket win check

### ISSUE THREAD

**Issue by yondonfu** [livepeer/protocol#268]
At the moment, the win check for a PM ticket is:

```
H(H(T), recipientRand) < T.winProb
```

where `T` is used as a random number given the assumption that `senderNonce` is a random number.

We can modify the win check to:

```
H(SENDER_SIG(H(T), recipientRand) < T.winProb
```

where `SENDER_SIG(H(T))` is used as a random number given the assumption that `senderNonce` is unique (not necessarily a random number). 

In this case, `senderNonce` can be a monotonic counter stored as a 8 byte unsigned integer. Every time a sender creates a new ticket it increments `senderNonce`. 

The benefits of this approach are:

- A smaller `senderNonce`. A random `senderNonce` would likely have to be stored as a large integer in order to avoid any requirements for the sender to keep track of already used `senderNonce` values. A monotonic `senderNonce` can be smaller and does not impose any additional memory requirements for the sender since it just needs to store a single integer that is incremented with each ticket. A smaller `senderNonce` also results in some small gas cost savings since `senderNonce` needs to be passed into TicketBroker in the transaction calldata (64 gas for each non-zero byte and 8 gas for each zero byte).
- Lower memory requirements for the recipient for replay attack defense. The recipient can just keep track of a single latest `senderNonce` value in the best case (and reject any lower values) or a few values in the worst case (in order to deal with potential out of order tickets).

**Comment by yondonfu** [comment]
Closed by #269

**Comment by j0sh** [comment]
> Such that if it loses or resets its state DB it would have no way of determining an acceptable senderNonce?

B would query for updated orchestrator info prior to each stream starting, and receive fresh PM parameters such as `recipientRand` with each query. With that, the  `senderNonce` can be reset for each stream.
